### PR TITLE
Machinery performance tweaks

### PR DESCRIPTION
**Pull request by Lohikar**
changes:
- Machines' `process()` has been separated from `auto_use_power()`.
- Lights no longer `process()` and instead use `auto_use_power()` for power calculations.
- Computers that didn't really need to `process()` no longer `process()`.
- Airlocks now use the scheduler to auto-close instead of `process()`ing.
- Fixed a bug where clicking on an AI Status display to set its status did not work.

known issues:
- [x] AI Status Displays cannot be changed.
- [x] Door close delays have a 2-second inaccuracy due to tick rate of `scheduler`, so shorted timing-circuits won't work correctly.

### CODE REVIEW

**Comment by Lohikar** [comment]
I have, they were calling it instead of calling Machinery's `inoperable()`. The calls have been replaced.

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -125,15 +124,31 @@
...
+		close_task.trigger_task_in(time)
+	else
+		schedule_task_with_source_in(time, src, /obj/machinery/door/proc/auto_close)
+
+/obj/machinery/door/proc/close_hatch_in(var/time = 5 SECONDS)
+	if (hatch_task)
+		// Update the time.
+		hatch_task.trigger_task_in(time)
+	else
+		schedule_task_with_source_in(time, src, /obj/machinery/door/proc/auto_close_hatch)
```
Comment (skull132) [inline code/game/machinery/doors/door.dm]: Shouldn't this go `hatch_task = schedule_task_with_source_in()`?

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -125,15 +124,31 @@
...
+/obj/machinery/door/proc/close_door_in(var/time = 5 SECONDS)
+	if (time < 2 SECONDS)	// Too short duration for the scheduler.
+		spawn(time)
+			src.auto_close()
+
+	else if (close_task)
+		// Update the time.
+		close_task.trigger_task_in(time)
+	else
+		schedule_task_with_source_in(time, src, /obj/machinery/door/proc/auto_close)
```
Comment (skull132) [inline code/game/machinery/doors/door.dm]: Same for this. close_task is never set. :ree:

File: code/game/machinery/machinery.dm [hunk 3]
```diff
@@ -141,7 +138,12 @@ Class Procs:
 	if(!(use_power || idle_power_usage || active_power_usage))
 		return PROCESS_KILL
 
-	return
+	return M_NO_PROCESS
+
+/obj/machinery/proc/get_process_type()
```
Comment (skull132) [inline code/game/machinery/machinery.dm]: Have you sweeped for machinery that toggles their idle_power_usage and so on?

**Pull request merged**